include/circt/Dialect/Ibis/IbisTypes.td

```diff
@@ -9,4 +9,68 @@
 #ifndef CIRCT_DIALECT_IBIS_IBISTYPES_TD
 #define CIRCT_DIALECT_IBIS_IBISTYPES_TD
 
+include "circt/Dialect/Ibis/IbisDialect.td"
+include "mlir/IR/AttrTypeBase.td"
+
+class IbisTypeDef<string name> : TypeDef<IbisDialect, name> { }
+
+def ClassRefType : IbisTypeDef<"ClassRef"> {
+  let mnemonic = "classref";
+  let parameters = (ins OptionalParameter<"mlir::FlatSymbolRefAttr">:$classRef);
+  let assemblyFormat = "`<` ($classRef^)?  `>`";
+  let description = [{
+    A reference to an Ibis class. May be either a reference to a specific
+    class (given a `$classRef` argument) or an opaque reference.
+  }];
+
+  let extraClassDeclaration = [{
+    bool isOpaque() {
+      return !getClassRef();
+    }
+  }];
+
+  let builders = [
+    // Builds an opaque (unresolved) class reference.
+    TypeBuilder<(ins), [{
+      return $_get($_ctxt, nullptr);
+    }]>,
+    TypeBuilder<(ins "StringAttr":$classRef), [{
+      return $_get($_ctxt, FlatSymbolRefAttr::get(classRef));
+    }]>
+  ];
+}
+
+def AnyClassRefType : Type<
```

Comment:
nit: "any" could be interpreted as including unknown classes or as being the Opaque version (as I first figured it was before reading the below). Might I suggest something like "known" or "transparent" or something to that effect?